app/_hub/kong-inc/response-ratelimiting/index.md

```diff
@@ -71,67 +71,85 @@ params:
     - name: limits.{limit_name}
       required: true
       value_in_examples: <SMS>
+      datatype:
       description: This is a list of custom objects that you can set, with arbitrary names set in the `{limit_name`} placeholder, like `config.limits.sms.minute=20` if your object is called "SMS".
     - name: limits.{limit_name}.second
       required: semi
+      datatype: number
       description: The amount of HTTP requests the developer can make per second. At least one limit must exist.
     - name: limits.{limit_name}.minute
       required: semi
       value_in_examples: 10
+      datatype: number
       description: The amount of HTTP requests the developer can make per minute. At least one limit must exist.
     - name: limits.{limit_name}.hour
       required: semi
+      datatype: number
       description: The amount of HTTP requests the developer can make per hour. At least one limit must exist.
     - name: limits.{limit_name}.day
       required: semi
+      datatype: number
       description: The amount of HTTP requests the developer can make per day. At least one limit must exist.
     - name: limits.{limit_name}.month
       required: semi
+      datatype: number
       description: The amount of HTTP requests the developer can make per month. At least one limit must exist.
     - name: limits.{limit_name}.year
       required: semi
+      datatype: number
       description: The amount of HTTP requests the developer can make per year. At least one limit must exist.
     - name: header_name
       required: false
       default: "`X-Kong-Limit`"
+      datatype: string
       description: The name of the response header used to increment the counters.
     - name: block_on_first_violation
-      required: false
+      required: true
       default: "`false`"
+      datatype: boolean
       description: A boolean value that determines if the requests should be blocked as soon as one limit is being exceeded. This will block requests that are supposed to consume other limits too.
     - name: limit_by
       required: false
       default: "`consumer`"
+      datatype: string
       description: "The entity that will be used when aggregating the limits: `consumer`, `credential`, `ip`. If the `consumer` or the `credential` cannot be determined, the system will always fallback to `ip`."
```

Comment:
Here in this plugin the `one_of` is represented in the description of the config field. Is it okay? Should we be adopting this convention in the plugins that are missing it such as https://github.com/Kong/docs.konghq.com/pull/2604?

Reply:
> Here in this plugin the `one_of` is represented in the description of the config field. Is it okay? Should we be adopting this convention in the plugins that are missing it such as #2604?

I think that is our best option really. We state in many places what the available options are, so this is an established convention.

Reply:
resolving convo per https://github.com/Kong/docs.konghq.com/pull/2617#issuecomment-784187024